### DISCUSSION TITLE
Xmidt release with tracing feature.

### DISCUSSION THREAD

**Discussion by Sachin4403** [xmidt-org/xmidt#43] [Q&A] [answered]
Hi team,

By when we can expect the release of the xmidt components with tracing feature is it blocked because of automatic  Instrumentation or it can be released with manual Instrumentation which is currently available in candlelight.

We are expecting these all components with tracing feature.
```
1. Tr1d1um
2. Scytale
3. Argus
4. Caduceus
5. Talaria
6. Petasos
7. Themis
```

**Comment by schmidtw** [comment, marked as answer]
Not directly solving the request, but here is a simple chart showing the release information so we can easily track completion.

### Tr1d1um
![GitHub Release Date](https://img.shields.io/github/release-date/xmidt-org/tr1d1um) [![GitHub release](https://img.shields.io/github/release/xmidt-org/tr1d1um.svg)](https://github.com/xmidt-org/tr1d1um/blob/main/CHANGELOG.md)

### Scytale
![GitHub Release Date](https://img.shields.io/github/release-date/xmidt-org/scytale) [![GitHub release](https://img.shields.io/github/release/xmidt-org/scytale.svg)](https://github.com/xmidt-org/scytale/blob/main/CHANGELOG.md)

### Argus
![GitHub Release Date](https://img.shields.io/github/release-date/xmidt-org/argus) [![GitHub release](https://img.shields.io/github/release/xmidt-org/argus.svg)](https://github.com/xmidt-org/argus/blob/main/CHANGELOG.md)

### Caduceus
![GitHub Release Date](https://img.shields.io/github/release-date/xmidt-org/caduceus) [![GitHub release](https://img.shields.io/github/release/xmidt-org/caduceus.svg)](https://github.com/xmidt-org/caduceus/blob/main/CHANGELOG.md)

### Talaria
![GitHub Release Date](https://img.shields.io/github/release-date/xmidt-org/talaria) [![GitHub release](https://img.shields.io/github/release/xmidt-org/talaria.svg)](https://github.com/xmidt-org/talaria/blob/main/CHANGELOG.md)

### Petasos
![GitHub Release Date](https://img.shields.io/github/release-date/xmidt-org/petasos) [![GitHub release](https://img.shields.io/github/release/xmidt-org/petasos.svg)](https://github.com/xmidt-org/petasos/blob/main/CHANGELOG.md)

### Themis
![GitHub Release Date](https://img.shields.io/github/release-date/xmidt-org/themis) [![GitHub release](https://img.shields.io/github/release/xmidt-org/themis.svg)](https://github.com/xmidt-org/themis/blob/main/CHANGELOG.md)

**Reply by Sachin4403** [reply]
Hello @schmidtw  Please mark this thread as resolved.

**Comment by joe94** [comment]
Hey @Sachin4403, the missing piece from the tracing feature is the ability to make it optional through configuration. We try to make features like this configurable and optional across our services. For instance, we (or other teams using XMiDT) might not enable it right away in our clusters.

**Reply by joe94** [reply]
I think we're getting close to the behavior we want in https://github.com/xmidt-org/candlelight/pull/23. I'll work on creating some sample code to show what I believe @johnabass is asking us to implement  🚀

**Reply by Sachin4403** [reply]
okay @joe94. I will wait for that.


Another thing which we should include in  release notes of 
```
ancla v0.1.3
tr1d1um v0.5.5
```
https://github.com/xmidt-org/ancla/issues/44

**Reply by joe94** [reply]
That change was included in `ancla/v0.1.2` https://github.com/xmidt-org/ancla/blob/main/CHANGELOG.md#v012

**Reply by joe94** [reply]
@Sachin4403, I spoke to @schmidtw today and I'll be helping out with the OpenTelemetry integration a bit more actively. Bear with us as we discuss how we plan to develop the rest of this feature. I'll try to keep you posted here and on the issues where we're doing the development.
So far, I'm seeing if I can get buy in from John on using the instrumentation packages mentioned in these issues for Tr1d1um: https://github.com/xmidt-org/tr1d1um/issues/198 https://github.com/xmidt-org/tr1d1um/issues/202

**Reply by Sachin4403** [reply]
okay @joe94, Please let me know if something is required from my end.

**Comment by joe94** [comment]
@Sachin4403, I merged the Tr1d1um PR 🎉  and added a project https://github.com/orgs/xmidt-org/projects/6 to keep track of our progress. 

I didn't add a lot of detail for each issue as I'm hoping Tr1d1um can serve as an example we can follow. We can always discuss details on each issue as things come up.

I started working on scytale so I assigned that issue to myself already. Please do the same on the issue you're currently working on.

**Reply by joe94** [reply]
To help visualize traces, feel free to use this branch https://github.com/xmidt-org/xmidt/pull/47 which shows how to export traces to Zipkin.

**Reply by Sachin4403** [reply]
Hi, @joe94  Will, you be doing it for **scytale** if yes then by when we can close it? If I have to do it I can open a PR tomorrow with all the changes according to the automatic instrumentation of the candlelight.

**Reply by joe94** [reply]
Yep, I'm working on the Scytale changes. I'm hoping to wrap it by tomorrow if not earlier👍
